
(env) PS C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG> python manage.py test inicio
Found 6 test(s).
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
E..EF.
======================================================================
ERROR: test_crear_oferta_empleo (inicio.tests.TestClienteFunctionality.test_crear_oferta_empleo)
Prueba la creación de una oferta de empleo.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG\inicio\tests.py", line 107, in test_crear_oferta_empleo
    self.assertRedirects(response, reverse('cliente_dashboard'))
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\testcases.py", line 515, in assertRedirects  
    redirect_response = response.client.get(
                        ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 1124, in get
    response = super().get(
               ^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 475, in get
    return self.generic(
           ^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 671, in generic
    return self.request(**r)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 1087, in request
    self.check_exception(response)
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 802, in check_exception     
    raise exc_value
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner    
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG\inicio\views.py", line 17, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG\inicio\views.py", line 96, in cliente_dashboard
    return render(request, 'cliente/dashboard.html', {'ofertas': ofertas, 'notificaciones': notificaciones})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\utils.py", line 114, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\utils.py", line 114, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\library.py", line 321, in render
    output = self.func(*resolved_args, **resolved_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\allauth\socialaccount\templatetags\socialaccount.py", line 21, in provider_login_url
    provider = adapter.get_provider(request, provider)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\allauth\socialaccount\adapter.py", line 212, in get_provider
    app = self.get_app(request, provider=provider, client_id=client_id)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\allauth\socialaccount\adapter.py", line 302, in get_app
    raise SocialApp.DoesNotExist()
allauth.socialaccount.models.SocialApp.DoesNotExist

======================================================================
ERROR: test_cliente_can_access_crear_oferta (inicio.tests.TestViewPermissions.test_cliente_can_access_crear_oferta)
Un Cliente DEBE poder acceder a la página de crear oferta.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG\inicio\tests.py", line 72, in test_cliente_can_access_crear_oferta
    response = self.client.get(reverse('crear_oferta'))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 1124, in get
    response = super().get(
               ^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 475, in get
    return self.generic(
           ^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 671, in generic
    return self.request(**r)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 1087, in request
    self.check_exception(response)
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\client.py", line 802, in check_exception
    raise exc_value
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG\inicio\views.py", line 17, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG\inicio\views.py", line 231, in crear_oferta
    return render(request, 'cliente/crear_oferta.html', {'form': form})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\utils.py", line 114, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\test\utils.py", line 114, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\django\template\library.py", line 321, in render
    output = self.func(*resolved_args, **resolved_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\allauth\socialaccount\templatetags\socialaccount.py", line 21, in provider_login_url
    provider = adapter.get_provider(request, provider)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\allauth\socialaccount\adapter.py", line 212, in get_provider
    app = self.get_app(request, provider=provider, client_id=client_id)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pedro\Desktop\Coding\JSAG\env\Lib\site-packages\allauth\socialaccount\adapter.py", line 302, in get_app
    raise SocialApp.DoesNotExist()
allauth.socialaccount.models.SocialApp.DoesNotExist

======================================================================
FAIL: test_postulante_cannot_access_crear_oferta (inicio.tests.TestViewPermissions.test_postulante_cannot_access_crear_oferta)
Un Postulante NO DEBE poder acceder a la página de crear oferta.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\pedro\Desktop\Coding\JSAG\JsAsesoriaG\inicio\tests.py", line 80, in test_postulante_cannot_access_crear_oferta
    self.assertRedirects(response, reverse('index'))
AssertionError: 302 != 200 : Couldn't retrieve redirection page '/': response code was 302 (expected 200)

----------------------------------------------------------------------
Ran 6 tests in 21.263s

FAILED (failures=1, errors=2)
Destroying test database for alias 'default'...